src/components/navigation/navigation-item/navigation-item.component.ts

```diff
@@ -0,0 +1,151 @@
+import {
+  Component,
+  Input,
+  ViewEncapsulation,
+  ContentChildren,
+  QueryList,
+  Renderer2,
+  ElementRef,
+  Optional,
+  SkipSelf,
+  AfterContentInit,
+  ContentChild,
+  Output,
+  AfterViewInit,
+  HostListener
+} from '@angular/core';
+import { Router, RouterLinkActive, NavigationEnd, RouterState, ActivatedRoute, UrlTree } from '@angular/router';
+import 'rxjs/add/operator/filter';
+
+@Component({
+    selector: '[ux-navigation-item]',
+    templateUrl: './navigation-item.component.html',
+    host: {
+        '[class.active]': 'active',
+        '[class.selected]': 'expanded',
+    },
+    encapsulation: ViewEncapsulation.None
+})
+export class NavigationItemComponent implements AfterViewInit {
+    @Input() header: string;
+    @Input() icon: string;
+    @Input() expanded: boolean = false;
+    @Input() link?: string;

```

Comment:
Remove the `?`. By default the value will be undefined unless the user specifies it.

Reply:
'?' removed.